react-native/react/actions/config/index.js

```diff
@@ -34,6 +34,15 @@ export function startup () {
         }
       })
     })
+
+    // Also call getCurrentStatus if the service goes away/comes back.
+    engine.listenOnConnect(() => {
+      engine.rpc('config.getCurrentStatus', {}, incomingMap, (error, status) => {
+        if (error) {
```

Comment:
this won't work , we need to dispatch also like above, else it does nothing if the status changes